components/Form/Form.tsx

```diff
@@ -32,15 +42,21 @@ const Form: React.FC<{
               <View style={style.field}>
                 <Input
                   placeholder={field.placeholder}
-                  onChangeText={val => onChange(val)}
+                  onChangeText={val => {
+                    const currentVal =
+                      field.keyboardType === 'numeric'
+                        ? val.replace(/[^0-9]/g, '')
+                        : val;
+                    onChange(currentVal);
+                  }}
                   keyboardType={field.keyboardType}
                   secureTextEntry={field.secureTextEntry}
                   inputValue={value}
                 />
```

Comment:
You should also use `returnKeyType` to let the native keyboard show the most appropriate button. This applies to all Inputs.